File: .tmp-run-checks.go
//go:build ignore

package main

import (
	"fmt"
	"os"
	"os/exec"
)

func run(cmd string, args ...string) int {
	fmt.Printf("===== COMMAND: %s =====\n", append([]string{cmd}, args...))
	c := exec.Command(cmd, args...)
	c.Dir = "/Volumes/Satechi/Developer/mockport"
	c.Stdout = os.Stdout
	c.Stderr = os.Stderr
	err := c.Run()
	code := 0
	if err != nil {
		if exitErr, ok := err.(*exec.ExitError); ok {
			code = exitErr.ExitCode()
		} else {
			code = 1
		}
	}
	fmt.Printf("EXIT_CODE=%d\n\n", code)
	return code
}

func main() {
	run("bash", "scripts/check-doc-links.sh")
	run("bash", "scripts/check-public-trust.sh")
}

File: .tmp-run-checks.mjs
const { execSync } = require('child_process');
const cwd = '/Volumes/Satechi/Developer/mockport';

function run(cmd) {
  console.log(`===== COMMAND: ${cmd} =====`);
  try {
    const out = execSync(cmd, { cwd, encoding: 'utf8', stdio: ['pipe', 'pipe', 'pipe'] });
    if (out) process.stdout.write(out);
    console.log('EXIT_CODE=0');
    return 0;
  } catch (e) {
    if (e.stdout) process.stdout.write(e.stdout);
    if (e.stderr) process.stderr.write(e.stderr);
    console.log(`EXIT_CODE=${e.status ?? 1}`);
    return e.status ?? 1;
  } finally {
    console.log();
  }
}

run('bash scripts/check-doc-links.sh');
run('bash scripts/check-public-trust.sh');
